diff -ur otags-3.06.2-/vi.ml otags-3.06.2/vi.ml
--- otags-3.06.2-/vi.ml	Tue Nov 12 21:20:55 2002
+++ otags-3.06.2/vi.ml	Tue Nov 12 21:21:07 2002
@@ -62,7 +62,7 @@
     let buffer = Buffer.create (String.length part_es) in
     String.iter (fun c ->
                    match c with
-                   | '.' | '$' | '*' | '[' | ']' | '/' | '\\' -> 
+                   | '$' | '/' | '\\' -> 
                        Buffer.add_char buffer '\\';
                        Buffer.add_char buffer c
                    | _ -> Buffer.add_char buffer c)
